exp/api/v1beta2/rosanetwork_types.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+// ROSANetworkSpec defines the desired state of ROSANetwork
+type ROSANetworkSpec struct {
+	// The name of the cloudformation stack under which the network infrastructure would be created
+	// +immutable
+	Name string `json:"name"`
```

Comment:
Not usual to have a field Name under spec. Better to change to vpcNetStackName or networkStackName
```suggestion
	vpcNetStackName string `json:"name"`
```